modules/integration/tests-common/admin-clients/src/main/java/org/wso2/bps/integration/common/clients/bpmn/ActivitiRestClient.java

```diff
@@ -0,0 +1,736 @@
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.methods.*;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.util.EntityUtils;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.apache.commons.logging.Log;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * This Class contains methods which uses the BPMN Rest Services to carryout BPMN Task
+ *
+ * @author WSO2
+ * @version 1.0
+ */
+public class ActivitiRestClient {
+
+	private static final Log log = LogFactory.getLog(ActivitiRestClient.class);
+	private final static String username = "admin";
+	private final static String password = "admin";
+	private final static String userClaim = "paul";
+	private final static String userDelegate = "will";
+	private int port;
+	private String hostname = "";
+	private String serviceURL = "";
+
+
+	public ActivitiRestClient(String portM, String hostnameM){
+		port = Integer.parseInt(portM);
+		hostname = hostnameM;
+		serviceURL = "http://" + hostnameM +":" + portM +"/bpmnrest/";
+	}
+
+	/**
+	 * Used put the thread in sleep until the tasks are generated
+	 */
+	public void waitForTaskGeneration() throws InterruptedException{
+		try {
+			Thread.sleep(1000);
+		}
+		catch (InterruptedException ie) {
+			log.error(" Thread Interuppted", ie);
+		}
+	}
+
+	//region DeploymentMethods
+	/**
+	 * This Method is used to deploy BPMN packages to the BPMN Server
+	 *
+	 * @param fileName The name of the Package to be deployed
+	 * @param filePath The location of the BPMN package to be deployed
+	 * @returns String array with status, deploymentID and Name
+	 * @throws java.io.IOException
+	 * @throws org.json.JSONException
+	 */
+	public String[] deployBPMNPackage(String filePath, String fileName) throws IOException,JSONException{
+		String url = serviceURL+"repository/deployments";
+
+		HttpHost target = new HttpHost(hostname,port,"http");
+
+		DefaultHttpClient httpClient = new DefaultHttpClient();
+		httpClient.getCredentialsProvider().setCredentials(
+				new AuthScope(target.getHostName(),target.getPort()),
+		        new UsernamePasswordCredentials(username,password)
+				);
+		try {
+
+			HttpPost httpPost = new HttpPost(url);
+
+			MultipartEntityBuilder builder = MultipartEntityBuilder.create();
+			builder.addBinaryBody("file", new File(filePath),
+			                      ContentType.MULTIPART_FORM_DATA,fileName);
+			HttpEntity multipart = builder.build();
+			httpPost.setEntity(multipart);
+			HttpResponse response = httpClient.execute(httpPost);
+			String status = response.getStatusLine().toString();
+			String responseData = EntityUtils.toString(response.getEntity());
+			JSONObject jsonResponseObject = new JSONObject(responseData);
```

Comment:
HTTP status codes should be available as constants from the library. Hence use those. 

https://hc.apache.org/httpclient-3.x/apidocs/constant-values.html